test/units/testsuite-99.sh

```diff
@@ -0,0 +1,127 @@
+set -eux
+set -o pipefail
+
+at_exit() {
+    # "Safety net" - check for any coredumps which might have not caused dfuzzer
+    # to stop & return an error (we need to do this now before truncating the
+    # journal)
+    local found_cd=0
+    while read -r exe; do
+        coredumctl info "$exe"
+        found_cd=1
+    done < <(coredumpctl -F COREDUMP_EXE | sort -u)
+    [[ $found_cd -eq 0 ]] || exit 1
+
+    # Limit the maximum journal size (because fuzzing)
+    journalctl --rotate --vacuum-size=16M
+}
+
+trap at_exit EXIT
+
+systemctl log-level info
+
+# TODO
+#   * check for possibly newly introduced buses?
+BUS_LIST=(
+    org.freedesktop.home1
+    org.freedesktop.hostname1
+    org.freedesktop.import1
+    org.freedesktop.locale1
+    org.freedesktop.login1
+    org.freedesktop.machine1
+    org.freedesktop.network1
+    org.freedesktop.oom1
+    org.freedesktop.portable1
+    org.freedesktop.resolve1
+    org.freedesktop.systemd1
+    org.freedesktop.timedate1
+    org.freedesktop.timesync1
+)
+
+SESSION_BUS_LIST=(
+    org.freedesktop.systemd1
+)
+
+# Maximum payload size generated by dfuzzer (in bytes) - default: 50K
+PAYLOAD_MAX=50000
+# Maximum memory size (in kilobytes) which a process can occupy before warning
+# is printed by dfuzzer - default: 512M
+MEMORY_MAX=524288
+# Tweak the maximum payload size if we're running under sanitizers, since
+# with larger payloads we start hitting reply timeouts
+if [[ -v ASAN_OPTIONS || -v UBSAN_OPTIONS ]]; then
+    PAYLOAD_MAX=10000 # 10K
+fi
+
+# Configure dfuzzer suppressions to filter out interfaces which might degrade
+# or straight up kill the test machine
+cat >/etc/dfuzzer.conf <<EOF
+[org.freedesktop.login1]
+Halt destructive
+HaltWithFlags destructive
+Hibernate destructive
+HibernateWithFlags destructive
+HybridSleep destructive
+HybridSleepWithFlags destructive
+KillSession destructive
+KillUser destructive
+LockSession destructive
+LockSessions destructive
+PowerOff destructive
+PowerOffWithFlags destructive
+Reboot destructive
+RebootWithFlags destructive
+ReleaseSession destructive
+ScheduleShutdown destructive
+Suspend destructive
+SuspendThenHibernate destructive
+SuspendThenHibernateWithFlags destructive
+SuspendWithFlags destructive
+TerminateSeat destructive
+TerminateSession destructive
+TerminateUser destructive
+
+[org.freedesktop.timedate1]
+SetLocalRTC destructive (screws up the RTC & system time)
+SetNTP destructive (disables systemd-timesyncd)
+EOF
+
+# Overmount /var/lib/machines with a size-limited tmpfs, as fuzzing
+# the org.freedesktop.machine1 stuff makes quite a mess
+mount -t tmpfs -o size=50M tmpfs /var/lib/machines
+
+# Create a log dir for dfuzzer to store potential failing test cases
+mkdir /dfuzzer-logs
+
+# Fuzz both the system and the session buses (where applicable)
+for bus in "${BUS_LIST[@]}"; do
+    echo "Bus: $bus (system)"
+    # Activate the name (if it's activatable) before fuzzing it
+    busctl --no-pager introspect "$bus" /
```

Comment:
Before I forget now that https://github.com/matusmarhefka/dfuzzer/pull/59 is merged it can be removed.

Reply:
> Another issue I ran into is that it can't seem to activate homed

As far as I can tell even with that PR included `dfuzzer` can't activate some services. My guess would be that it worked on CentOS CI because it enabled those services somehow somewhere. In my containers I have to run `systemctl enable systemd-systemd` explicitly to make it work.

Reply:
Yeah, it's explicitly pulled in in the testsuite service file (https://github.com/systemd/systemd/pull/22547/commits/a98a0a89f5e1f1f627131cff6c528bd1856bf31c?diff=unified&w=0#diff-db05ad3d53fd2ac714b4a714d249f3e237090416860cdca964a1dfbafad429beR4-R7).

Reply:
I ran the test with `make clean setup run` so it theory that part should have been included. I'm not sure why it didn't work out.

Reply:
I wonder if this is not a bug on systemd's side, since:

```
dbus-broker-launch[1524]: Activation request for 'org.freedesktop.home1' failed: The systemd unit 'dbus-org.freedesktop.home1.service' could not be found.
```

and indeed, there's no dbus service file for homed:

```
$ ls /usr/lib/systemd/system | grep dbus-
dbus-broker.service
dbus-daemon.service
dbus-org.freedesktop.hostname1.service
dbus-org.freedesktop.import1.service
dbus-org.freedesktop.locale1.service
dbus-org.freedesktop.login1.service
dbus-org.freedesktop.machine1.service
dbus-org.freedesktop.oom1.service
dbus-org.freedesktop.portable1.service
dbus-org.freedesktop.timedate1.service
```

since we don't explicitly install it:

https://github.com/systemd/systemd/blob/main/units/meson.build#L228-L233

Reply:
If it isn't intentional I think it's safe to assume `dfuzzer` has found another issue

Reply:
After a quick chat with Lennart, this is actually expected, since in case of the "dynamic" units the dbus service symlink is created when the respective systemd unit is enabled, e.g.:

```
# systemctl enable systemd-homed
Created symlink /etc/systemd/system/dbus-org.freedesktop.home1.service → /usr/lib/systemd/system/systemd-homed.service.
Created symlink /etc/systemd/system/multi-user.target.wants/systemd-homed.service → /usr/lib/systemd/system/systemd-homed.service.
Created symlink /etc/systemd/system/systemd-homed.service.wants/systemd-homed-activate.service → /usr/lib/systemd/system/systemd-homed-activate.service.
Created symlink /etc/systemd/system/sockets.target.wants/systemd-userdbd.socket → /usr/lib/systemd/system/systemd-userdbd.socket.
```

So I guess the test should be tweaked to either do `systemctl preset` or enable the required systemd services.